modules/aws/src/smithy4s/aws/AwsCredentialsProvider.scala

```diff
@@ -36,24 +39,33 @@ object AwsCredentialsProvider {
 
   def default[F[_]](
       httpClient: SimpleHttpClient[F]
-  )(implicit F: Temporal[F]): Resource[F, F[AwsCredentials]] = {
-    Resource
-      .eval(fromEnv[F])
-      .map(F.pure)
-      .orElse(refreshing[F](fromECS(httpClient)))
+  )(implicit F: Async[F]): Resource[F, F[AwsCredentials]] = {
+    val env = Resource.eval(fromEnv[F]).map(F.pure)
+    val fromDisk = Resource.eval(loadFromDisk[F]).map(F.pure)
+    env
+      .orElse(fromDisk)
       .orElse(refreshing[F](fromEC2(httpClient)))
   }
 
   def fromEnv[F[_]](implicit F: MonadThrow[F]): F[AwsCredentials] = {
-    val either = for {
+    val either: Either[Throwable, AwsCredentials] = for {
       keyId <- SysEnv.envValue(AWS_ACCESS_KEY_ID)
       accessKey <- SysEnv.envValue(AWS_SECRET_ACCESS_KEY)
       session = SysEnv.envValue(AWS_SESSION_TOKEN).toOption
     } yield AwsCredentials.Default(keyId, accessKey, session)
-    either match {
-      case Right(value) => F.pure(value)
-      case Left(t)      => F.raiseError(t)
-    }
+    either.liftTo[F]
+  }
+
+  def loadFromDisk[F[_]](implicit F: Sync[F]): F[AwsCredentials] = {
```

Comment:
let's just call this `fromDisk` (drop the `load`, it's implied), taking a `Profile` for the profile as a parameter. Then create a `getProfileFromEnv[F[_]]` method (profiles may be fetched from elsewhere, a config file or whatnot).  Then wire these methods in the `default` method. 


Also, let's make a newtype for `Profile` (look at `AwsRegion`) for inspiration.

Reply:
I encoded it as `def getProfileFromEnv: Option[String]` because `F` is not required. Unless we change `SysEnv.envValue` to return a `F`

Reply:
CE went with `F` for env reading, YMMV though